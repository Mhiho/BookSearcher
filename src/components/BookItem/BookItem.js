import React from 'react';
import classes from './BookItem.module.scss';

const bookItem = ({book}) => (
  <div className={classes.item}>
    <h3 className={classes.title}>{book.title ? book.title : "Brak tytu≈Çu"}</h3>
    <div className={classes.frame}>
    <img src={book.thumbnail} alt="photo"/>
    </div>
    <div className={classes.authors}>{book.authors ? book.authors.map(author=>( <h4>{author}</h4> )) : "Brak autora"}</div>
    <p className={classes.description}>{book.description? book.description.replace(/^(.{100}[^\s]*).*/, "$1") : "Brak opisu."}</p>
  </div>
)

export default bookItem;
