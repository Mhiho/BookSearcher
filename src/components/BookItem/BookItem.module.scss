.item {
  width: 33%;
}
.title {
  height: 20vh;
}
.title, img, .description, .authors {
  display: flex;
  justify-content: center;
  margin: 3% 5% 3% 5%;
}
.description {
  height: 10vh;
}
img {
  height: 25vh;
}
.frame {
  display: flex;
  justify-content: center;
  align-items: center;
}
.authors {
  height: 10vh;
}
