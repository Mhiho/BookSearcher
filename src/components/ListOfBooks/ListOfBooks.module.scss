.row {
  width: 100%;
  align-items: center;
  display: flex;
  align-items: center;
  justify-content: center;
  flex-wrap: wrap;
}
