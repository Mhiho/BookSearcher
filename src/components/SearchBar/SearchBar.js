import React, {Component} from 'react';
import books from 'google-books-search';
import cut from 'cut-substring';

const API_KEY = 'AIzaSyAbv1wv1o5lzTgBCniCGLMaW1pVthKgzwY';

var options = {
    key: API_KEY,
    field: 'title',
    offset: 0,
    limit: 10,
    type: 'books',
    order: 'relevance',
    lang: 'en'
};

class SearchBar extends Component {

  state = {
    value: "",
    books: [],
  }

  handleChange(event) {
    this.setState({value: event.target.value})
  }

  handleSubmit(event) {
    let self = this;
    books.search("Professional JavaScript for Web Developers", options, function(error, results, apiResponse) {
    if ( ! error ) {
      self.setState({books: results})
    } else {
        console.log(error);
    }
});

  event.preventDefault();
  }

  render() {
    const books = this.state.books;
    console.log(books);
    console.log(this.state.books)
    return (
      <div>
        <form onSubmit={this.handleSubmit.bind(this)}>
          <input
            onChange={this.handleChange.bind(this)}
            type="search"
            value={this.state.value}
            />
            <input
            type="submit"
            value="search"
            />
          </form>
          <div>
            {

              books.map(book=>{
              const descr = book.description ? book.description.replace(/^(.{100}[^\s]*).*/, "$1") : book.description
              return (
              <div key={book.id}>
                <img src={book.thumbnail} alt="link"/>
                <h3>{book.title}</h3>
                <p>
                {descr}
                </p>
              </div>
            )
            })}
          </div>
      </div>
    )
  }

}

export default SearchBar;
