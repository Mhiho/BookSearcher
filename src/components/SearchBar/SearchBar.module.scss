.row {
  width: 100%;
  align-items: center;
  height: 50px;
  display: flex;
  align-items: center;
  justify-content: center;
}

.input, .inputSubmit {
  width: 100%;
  display: flex;
  height: 32px;
}
.input {
  border-radius: 5px 0 0 5px;
}
.inputSubmit {
  border-radius: 0 5px 5px 0;
  display: flex;
  justify-content: center;
  background: #eee;
  color: cornflowerblue;
  border: 1px solid darkgrey;
  font-weight: 900;
}
.inputSubmit:hover {
  background: darken(#eee, 7%);
  color: #9b6a12;
}
form {
  display: flex;
}
