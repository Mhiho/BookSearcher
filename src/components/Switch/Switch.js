import React, {Component} from 'react';

class Switch extends Component {

  state = {
    option: 'title'
  }
  handleChange(event) {
    this.setState({option: event.target.value})
    this.props.switch(this.state.option)

  }

  render(){
    return (
      <form>
        <select value={this.state.option} onChange={this.handleChange.bind(this)}>
          <option value="author">author</option>
          <option value="title">title</option>
          <option value="language">language</option>
          <option value="publishedDate">publishedDate</option>
          </select>
      </form>
    )
  }
}

export default Switch;
