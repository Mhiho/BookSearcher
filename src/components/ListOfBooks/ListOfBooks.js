import React from 'react';
import BookItem from '../BookItem/BookItem';
import classes from './ListOfBooks.module.scss';
import ErrorBoundary from '../ErrorBoundary';
import _ from 'lodash';

const ListOfBooks = (props) => {
    const list = _.map(props.books,book=>(
      <BookItem
        key={book.id}
        className={classes.item}
        title={book.title}
        img={book.thumbnail}
        authors={book.authors}
        description={book.description}
      />
    ))
    console.log(_.map(props.books, book => book.id))
    return (
      <ErrorBoundary className={classes.row}>
       {list}
      </ErrorBoundary>
    )
}


export default ListOfBooks;
