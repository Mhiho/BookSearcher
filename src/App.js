import React, { Component } from 'react';
import classes from './App.module.scss';
import books from 'google-books-search';
import SearchBar from './components/SearchBar/SearchBar';
import ListOfBooks from './components/ListOfBooks/ListOfBooks';



class App extends Component {
  constructor(props) {
    super(props);

    this.state = {
      books: [],
      field: 'language',
      lang: 'en'
    };

  const API_KEY = 'AIzaSyAbv1wv1o5lzTgBCniCGLMaW1pVthKgzwY';
  let options = {
      key: API_KEY,
      field: this.state.field,
      offset: 0,
      limit: 10,
      type: 'books',
      order: 'relevance',
      lang: 'en'
  };
  }
  bookSearch(value) {
    let self = this;
    books.search(value, this.options, function(error, results, apiResponse) {
    if ( ! error ) {
      self.setState({books: results})
    } else {
        console.log(error);
    }
});
  }

  render() {
    console.log(this.state.field)
    return (
      <div>
        <SearchBar bookSearch={this.bookSearch.bind(this)}/>
        <ListOfBooks books={this.state.books} />
      </div>
    );
  }
}

export default App;
