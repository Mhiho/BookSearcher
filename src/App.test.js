import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import SearchBar from './components/SearchBar/SearchBar'
import BookItem from './components/BookItem/BookItem';
import ListOfBooks from './components/ListOfBooks/ListOfBooks';
import Switch from './components/Switch/Switch';
import { create } from "react-test-renderer";

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
});


  test("searchBar", () => {
    const component = create(<SearchBar />);
    expect(component.toJSON()).toMatchSnapshot();
  });
  test("bookItem", () => {
    const component = create(<BookItem />);
    expect(component.toJSON()).toMatchSnapshot();
  });
  test("ListOfBooks", () => {
    const component = create(<ListOfBooks />);
    expect(component.toJSON()).toMatchSnapshot();
  });
  test("switch", () => {
    const component = create(<Switch />);
    expect(component.toJSON()).toMatchSnapshot();
  });
