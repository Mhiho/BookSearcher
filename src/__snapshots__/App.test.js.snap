// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListOfBooks 1`] = `null`;

exports[`bookItem 1`] = `
Array [
  <h3
    className="title"
>
    Brak tytu≈Çu
</h3>,
  <div
    className="frame"
>
    <img
        alt="pff"
        src={undefined}
    />
</div>,
  <div
    className="authors"
>
    Brak autora
</div>,
  <p
    className="description"
>
    Brak opisu.
</p>,
]
`;

exports[`searchBar 1`] = `
<div
  className="row"
>
  <form
    onSubmit={[Function]}
  >
    <input
      className="input"
      onChange={[Function]}
      placeholder="type here"
      type="search"
      value=""
    />
    <input
      className="inputSubmit"
      type="submit"
      value="search"
    />
  </form>
</div>
`;

exports[`switch 1`] = `
<div
  className="selectCont"
>
  <select
    onChange={[Function]}
    value="title"
  >
    <option
      value="author"
    >
      author
    </option>
    <option
      value="title"
    >
      title
    </option>
  </select>
</div>
`;
